vendor/bat-native-ads/src/bat/ads/internal/account/confirmations/confirmations.h

```diff
@@ -35,15 +34,17 @@ class Confirmations final : public RedeemUnblindedTokenDelegate {
   explicit Confirmations(privacy::TokenGeneratorInterface* token_generator);
   ~Confirmations() override;
 
-  void AddObserver(ConfirmationsObserver* observer);
-  void RemoveObserver(ConfirmationsObserver* observer);
+  void set_delegate(ConfirmationsDelegate* delegate) {
+    DCHECK_EQ(delegate_, nullptr);
+    delegate_ = delegate;
+  }
 
   void Confirm(const TransactionInfo& transaction);
 
   void ProcessRetryQueue();
 
  private:
-  base::ObserverList<ConfirmationsObserver> observers_;
+  ConfirmationsDelegate* delegate_ = nullptr;
```

Comment:
we should use `raw_ptr<>` for raw pointers: https://github.com/brave/brave-core/commit/0e460f97acc16825c4547a741647952e25447ac7

Reply:
Thanks. I have raised https://github.com/brave/brave-browser/issues/20805 to fix this throughout Brave Ads